### PR TITLE
Scoreboard

### PR DESCRIPTION
**Pull request by Saad22386**
[usbdev]  usbdev scoreboard

this PR contains integration of predictor in usbdev scoreboard 

@jdonjdon

### CODE REVIEW

**Comment by jdonjdon** [comment]
> @jdonjdon why these checks are failing.

023-12-21T15:56:37.7543331Z * `Error-[SFCOR] Source file cannot be opened` has 1 failures:
2023-12-21T15:56:37.7543659Z     * Test cover_reg_top has 1 failures.
2023-12-21T15:56:37.7543930Z         * cover_reg_top\
2023-12-21T15:56:37.7544402Z           Line 1262, in log /azp/agent/_work/1/s/scratch/HEAD/usbdev-sim-vcs/cover_reg_top/build.log
2023-12-21T15:56:37.7544643Z 
2023-12-21T15:56:37.7545029Z                 Error-[SFCOR] Source file cannot be opened
2023-12-21T15:56:37.7545378Z                   Source file "usbdev_packetiser.sv" cannot be opened for reading due to 'No
2023-12-21T15:56:37.7545700Z                   such file or directory'.
2023-12-21T15:56:37.7545989Z                   Please fix above issue and compile again.
2023-12-21T15:56:37.7546306Z                   "../src/lowrisc_dv_usbdev_env_0.1/usbdev_env_pkg.sv", 56


This happens due to "usbdev_packetiser.sv" is not added to 
https://github.com/lowRISC/opentitan/blob/master/hw/ip/usbdev/dv/env/usbdev_env.core

**Comment by Saad22386** [comment]
> > @jdonjdon why these checks are failing.
> 
> 023-12-21T15:56:37.7543331Z * `Error-[SFCOR] Source file cannot be opened` has 1 failures: 2023-12-21T15:56:37.7543659Z * Test cover_reg_top has 1 failures. 2023-12-21T15:56:37.7543930Z * cover_reg_top 2023-12-21T15:56:37.7544402Z Line 1262, in log /azp/agent/_work/1/s/scratch/HEAD/usbdev-sim-vcs/cover_reg_top/build.log 2023-12-21T15:56:37.7544643Z 2023-12-21T15:56:37.7545029Z Error-[SFCOR] Source file cannot be opened 2023-12-21T15:56:37.7545378Z Source file "usbdev_packetiser.sv" cannot be opened for reading due to 'No 2023-12-21T15:56:37.7545700Z such file or directory'. 2023-12-21T15:56:37.7545989Z Please fix above issue and compile again. 2023-12-21T15:56:37.7546306Z "../src/lowrisc_dv_usbdev_env_0.1/usbdev_env_pkg.sv", 56
> 
> This happens due to "usbdev_packetiser.sv" is not added to https://github.com/lowRISC/opentitan/blob/master/hw/ip/usbdev/dv/env/usbdev_env.core

It's means i have to add other classes as well that involved in usbdev predictor.
or just i have to add usbdev_packetiser.sv?

**Comment by jdonjdon** [comment]
> > > @jdonjdon why these checks are failing.
> > 
> > 
> > 023-12-21T15:56:37.7543331Z * `Error-[SFCOR] Source file cannot be opened` has 1 failures: 2023-12-21T15:56:37.7543659Z * Test cover_reg_top has 1 failures. 2023-12-21T15:56:37.7543930Z * cover_reg_top 2023-12-21T15:56:37.7544402Z Line 1262, in log /azp/agent/_work/1/s/scratch/HEAD/usbdev-sim-vcs/cover_reg_top/build.log 2023-12-21T15:56:37.7544643Z 2023-12-21T15:56:37.7545029Z Error-[SFCOR] Source file cannot be opened 2023-12-21T15:56:37.7545378Z Source file "usbdev_packetiser.sv" cannot be opened for reading due to 'No 2023-12-21T15:56:37.7545700Z such file or directory'. 2023-12-21T15:56:37.7545989Z Please fix above issue and compile again. 2023-12-21T15:56:37.7546306Z "../src/lowrisc_dv_usbdev_env_0.1/usbdev_env_pkg.sv", 56
> > This happens due to "usbdev_packetiser.sv" is not added to https://github.com/lowRISC/opentitan/blob/master/hw/ip/usbdev/dv/env/usbdev_env.core
> 
> It's means i have to add other classes as well that involved in usbdev predictor. or just i have to add usbdev_packetiser.sv?

Add usbdev_packetiser.sv to usbdev_env.core

**Comment by Saad22386** [comment]
@jdonjdon it's still failing.

**Comment by jdonjdon** [comment]
> @jdonjdon it's still failing.

I am not sure how this is pass on your end.
CI fails because usb20_item declared as abstract
(https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/usb20_agent/usb20_item.sv;drc=5309af625e402327f8f6ff95cbe3bb0477226960;l=12)

you need to remove `virtual` keyword from 
(https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/usb20_agent/usb20_item.sv;drc=5309af625e402327f8f6ff95cbe3bb0477226960;l=12)

**Comment by Saad22386** [comment]
> > @jdonjdon it's still failing.
> 
> I am not sure how this is pass on your end. CI fails because usb20_item declared as abstract (https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/usb20_agent/usb20_item.sv;drc=5309af625e402327f8f6ff95cbe3bb0477226960;l=12)
> 
> you need to remove `virtual` keyword from (https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/usb20_agent/usb20_item.sv;drc=5309af625e402327f8f6ff95cbe3bb0477226960;l=12)

Another PR was done by @MubashirSaleem775 yesterday in which virtual keyword was removed. 
But that PR is still under review.

File: hw/ip/usbdev/dv/env/usbdev_scoreboard.sv [hunk 1]
```diff
@@ -9,7 +9,14 @@ class usbdev_scoreboard extends cip_base_scoreboard #(
...
 
-  // local variables
+  usbdev_packetiser  m_packetiser;
+  usbdev_TransactionManager m_usbdev_trans;
+  usbdev_pkt_manager m_pkt_manager;
+  usb20_item m_usb20_item;
+
+  // Arrays to store actual packet and expected packet
+  bit expected_pkt[];
+  bit actual_pkt[];
```
Comment (jdonjdon) [inline hw/ip/usbdev/dv/env/usbdev_scoreboard.sv:19]: Line 29, I don't see you use usb20_q anywhere.
Is this for the future usage?

File: hw/ip/usbdev/dv/env/usbdev_scoreboard.sv [hunk 2]
```diff
@@ -36,17 +47,33 @@ class usbdev_scoreboard extends cip_base_scoreboard #(
     super.run_phase(phase);
     fork
       process_usb20_fifo();
-    join_none
+    join
```
Comment (jdonjdon) [inline hw/ip/usbdev/dv/env/usbdev_scoreboard.sv:50]: given `process_usb20_fifo()` has forever loop, why did you change from join_non to join?

Comment (Saad22386) [reply hw/ip/usbdev/dv/env/usbdev_scoreboard.sv:50]: Yes

File: hw/ip/usbdev/dv/env/usbdev_scoreboard.sv (hunk 1 above)
Comment (Saad22386) [reply hw/ip/usbdev/dv/env/usbdev_scoreboard.sv:19]: Yes

File: hw/ip/usbdev/dv/env/usbdev_scoreboard.sv (hunk 2 above)
Comment (Saad22386) [reply hw/ip/usbdev/dv/env/usbdev_scoreboard.sv]: This was done unintentionally during debugging .
Now I have changed it to join_none.

**Review by jdonjdon — APPROVED** [review]

**Comment by Saad22386** [comment]
@jdonjdon

**Review by jdonjdon — APPROVED** [review]
